### PR TITLE
feat(cli): [R8] add local digest command

### PR DESCRIPTION
**Pull request by ebarti**
## Summary
- Adds `jobhunter digest` for local digest reads and explicit acknowledgement.
- Keeps digest reads passive; `digest_state` advances only through `--acknowledge` or the panel acknowledge action.
- Records `DigestReviewed` events on acknowledgement and documents the CLI/user flow.

## Stack
- Previous: #291 (`feat/r8-p4-digest-panel`)
- Final PR in this stack.

## Validation
Full stack validation was run from this branch:
- `pnpm check`
- `pnpm test`
- `pnpm --filter @jobhunter/web test-d`
- `pnpm --filter @jobhunter/web test`
- `pnpm --filter @jobhunter/web e2e -- tests/dashboard.spec.ts`
- `pnpm web:storybook:build`
- `pnpm docs:build`
- `git diff --check`

Focused digest/CLI validation:
- `uv --project workers/automation run --extra dev pytest -q workers/automation/tests/test_digest_parity.py workers/automation/tests/test_digest_cli.py`
- QA verified `GET /v1/digest` does not advance `digest_state` and `POST /v1/digest/acknowledge` records `DigestReviewed`.

Review gates:
- pr-reviewer: Gate: PASS
- QA: Gate: PASS

Known follow-up
- pr-reviewer reported a non-blocking Medium: follow-up digest counts are outcome-derived while the current deep link targets the existing applied Jobs view. No Blocker or High findings remain.

### CODE REVIEW

File: workers/automation/src/jobhunter/cli.py [hunk 1]
```diff
@@ -1134,6 +1134,173 @@ def job(
...
+    acknowledge_payload = None
+    if acknowledge:
+        acknowledge_payload = acknowledge_digest(
+            conn,
+            acknowledged_at=str(digest_payload["generatedAt"]),
+        )
+
+    if json_output:
+        payload = (
+            {"digest": digest_payload, "acknowledge": acknowledge_payload}
```
Comment (ebarti) [inline workers/automation/src/jobhunter/cli.py:1172]: PR Reviewer - pr-reviewer

Severity: Low
Gate: Pass
Evidence: cli.py:1170-1176. With --json alone, the top-level JSON IS the digest object. With --json --acknowledge, the top level becomes {"digest": ..., "acknowledge": ...}. Confirmed by test_digest_cli.py::test_digest_cli_json_acknowledges_generated_digest, which reads payload["digest"]["newMatches"], whereas a plain --json consumer reads newMatches at top level.
Reproduction: Reasoned from code and existing tests.

The machine-readable --json contract changes its top-level shape depending on whether --acknowledge is passed. A script such as `jobhunter digest --json | jq .newMatches.count` silently breaks when --acknowledge is added (the field moves to .digest.newMatches.count). For a flag explicitly labeled "machine-readable JSON", an unstable top-level shape is a scriptability wart.

Required fix: Emit one stable top-level shape regardless of --acknowledge (for example, always {"digest": ..., "acknowledge": <object|null>}), or document both shapes in --help.

File: workers/automation/src/jobhunter/cli.py [hunk 2]
```diff
@@ -1134,6 +1134,173 @@ def job(
...
+            if acknowledge_payload
+            else digest_payload
+        )
+        console.print_json(data=payload)
+        return
+
+    since = digest_payload.get("since") or "first run"
```
Comment (ebarti) [inline workers/automation/src/jobhunter/cli.py:1179]: PR Reviewer - pr-reviewer

Severity: Medium
Gate: Pass
Evidence: cli.py:1179 (`since = digest_payload.get("since") or "first run"`) plus the zero-count rows in _render_digest_table and the None-return in _render_digest_links_table. All three CLI tests in test_digest_cli.py stub build_digest to return _sample_digest(), which always has a non-null since and non-zero counts. No CLI test exercises a first-run/empty digest (since=None, all counts 0, empty deepLinks). The task explicitly requested an "empty digest" test case, and it is absent.
Reproduction: Reasoned from code and tests. I traced _render_digest_table, _budget_note, _blocked_source_note, and _render_digest_links_table with an empty payload: they all fall back to safe defaults, so this is a coverage gap, not a latent crash.

The first-run / empty-state rendering path (the "first run" fallback, zero-count rows, and the suppressed deep-links table) is entirely untested at the CLI layer. A future regression in zero-state rendering would ship undetected.

Required fix: Add a CLI unit test that stubs build_digest to a first-run/empty digest (since=None, all counts 0, deepLinks empty) and asserts the command exits 0, prints "first run", and renders zeros without raising.

**Review by ebarti — COMMENTED** [review]
PR Reviewer - pr-reviewer — Summary

Gate: PASS (Blocker=0, High=0, Medium=1, Low=1)

Reviewed PR #292 (feat/r8-p5-digest-cli) against base feat/r8-p4-digest-panel at head 6be1df91. Merge-base 01abc0dc; diff is clean (9 files, +512/-3).

What was implemented: a read-only, on-demand `jobhunter digest` Typer command (P5 of the saved-views/digest plan). It reads local SQLite directly via the existing Python build_digest, prints a rich table plus deep-link table, supports --json and --min-fit-score, and --acknowledge advances the digest_state watermark and records a DigestReviewed event. No Temporal schedule was added.

Architecture and parity (the core risks) — all clear:
- Dual-source is plan-authorized. The plan (lines 452-457) explicitly says the CLI reads SQLite directly because the CLI has no read-side RPC and the API may be down, and mandates a parity test. build_digest / acknowledge_digest are the Python mirror of the TS read-model.ts implementation.
- Parity is real and enforced on both sides. apps/api/test/digest.test.ts and workers/automation/tests/test_digest_parity.py both bind to the same shared fixture packages/domain-types/test/fixtures/daily_digest_parity.json and assert the same expected block, including acknowledge behavior.
- Acknowledge is the SAME mechanism, not a side door. Python acknowledge_digest writes digest_state and a DigestReviewed row into job_events with byte-identical payload shape/order to the TS path; DigestReviewed is a registered domain event with a frontend handler (invalidation-router.ts:191), so the CLI write flows through the same SSE invalidation path as the web panel. Watermark is monotonic and bounded to now on both sides.
- Threshold parity: read_min_fit_score reads min_fit_score from the same dashboard.json the TS API writes, mirroring the sibling read_daily_budget_usd — no drift.
- Runtime honesty: the command needs no Temporal server, worker, or API; _bootstrap only does local env/DB/projection work (projection refresh is wrapped in try/except); init_db means a fresh DB yields zero-count output, not a crash. It does not trip the work-starting path.
- Sensitive-data posture: stdout only, no file writes, budget shows the same USD figures as the web digest, no secrets or resume/cover content.
- Scope + docs + conventions: diff stays within workers/automation plus docs; README command tree + local-only safety note, docs/user/normal-flows.md, docs/architecture/storage.md, and docs/local-reliability-qa.md all updated; Conventional Commit title with [R8].

Findings (both non-gating):

| # | Severity | Location | Finding |
|---|----------|----------|---------|
| 1 | Low | cli.py:1172 | --json top-level shape changes when --acknowledge is passed (bare digest vs {digest, acknowledge}); unstable machine-readable contract. |
| 2 | Medium | cli.py:1179 | First-run/empty digest rendering (since=None, zero counts, empty deepLinks) is untested at the CLI layer; the requested empty-digest case is absent. Code is safe-by-default (traced), so this is a coverage gap, not a crash. |

Verification (run in isolated worktree at head):
- uv --project workers/automation run --extra dev pytest -q workers/automation/tests/test_digest_cli.py workers/automation/tests/test_digest_parity.py -> 5 passed.
- uv --project workers/automation run --extra dev ruff check . -> All checks passed.
- uv --project workers/automation run --extra dev pytest -q (full) -> run 1: 1 failed (test_llm_client.py::test_chat_retries_on_negative_retry_after), 1821 passed; run 2: 1822 passed. The failure is a pre-existing FLAKY test-isolation issue unrelated to this PR: that test monkeypatches the global time.sleep and a leaked background thread (Temporal heartbeat/retry, visible in base-run logs) sleeps during the test window. llm.py/test_llm_client.py are untouched by this PR; base full suite is green; the 4 new digest tests pass deterministically in isolation, minimal-repro, and both full runs. Not gating.
- jobhunter digest --help renders correctly with the [1<=x<=10] range constraint.

Residual risk:
- Budget-read parity between Python read_spend_budget_status and TS readLlmSpendHealth is not exercised by the parity fixture (it stubs budget on both sides). The two readers use the same dashboard.json budget key and identical derivation, and the CLI is the first caller of build_digest that uses the live budget reader; the pre-existing UTC-day vs local-day note in the plan applies. Pre-existing, not introduced here.
- The full Python suite has a latent flaky isolation defect (global time.sleep monkeypatch + leaked worker threads); worth a follow-up but outside this PR.
